src/clients/SpokePoolClient.ts

```diff
@@ -732,8 +746,10 @@ export class SpokePoolClient extends BaseAbstractClient {
         const rawFill = spreadEventWithBlockNumber(event) as FillWithBlock;
         const fill: FillWithBlock = {
           ...rawFill,
-          blockTimestamp: blocks[event.blockNumber].timestamp,
         };
+        if (blocks !== undefined) {
```

Comment:
```suggestion
        if (isDefined(blocks)) {
```
This also checks for blocks === null which isn't necessarily required but it's a good way to round off the falsy values.